tart2ms/tart2ms.py

```diff
@@ -690,43 +778,147 @@ def __twelveball(direction):
-                                  freq=spw_chan_freqs[spw_id],
-                                  pos=np.rad2deg(centroid_direction[0, :]),
-                                  uvw=uvw_data,
-                                  refdir=np.rad2deg(zenith_directions),
-                                  field_ids=map_row_to_zendir)
-                model_data = rephased_data
+                                dask_data,
+                                dtype=dask_data.dtype,
+                                chunks=dask_data.chunks,
+                                # kwargs for rephase
+                                freq=spw_chan_freqs[spw_id],
+                                pos=np.rad2deg(centroid_direction[0, :]),
+                                uvw=uvw_data,
+                                refdir=np.rad2deg(zenith_directions),
+                                field_ids=map_row_to_zendir)
+                dask_data = rephased_data
+                rephased_data = da.empty_like(dask_data)
+                if fill_model:
+                    rephased_data = \
+                        da.map_blocks(rephase,
+                                    model_data,
+                                    dtype=model_data.dtype,
+                                    chunks=model_data.chunks,
+                                    # kwargs for rephase
+                                    freq=spw_chan_freqs[spw_id],
+                                    pos=np.rad2deg(centroid_direction[0, :]),
+                                    uvw=uvw_data,
+                                    refdir=np.rad2deg(zenith_directions),
+                                    field_ids=map_row_to_zendir)
+                    model_data = rephased_data
+            elif centroid_direction.shape[0] == len(obstime):
+                LOGGER.info(f"Per user request: Rephase data to special field {phase_center_policy.replace('rephase-','')} per timestamp")
+                rephased_data = da.zeros_like(dask_data)
+                subfields = np.unique(map_row_to_zendir)
+                p = progress(
+                    "Phasing data", max=subfields.size)
+                for sfi in subfields:
+                    sel = map_row_to_zendir == sfi
+                    rephased_data += \
+                        da.map_blocks(rephase,
+                                    dask_data,
+                                    dtype=dask_data.dtype,
+                                    chunks=dask_data.chunks,
+                                    # kwargs for rephase
+                                    freq=spw_chan_freqs[spw_id],
+                                    pos=np.rad2deg(centroid_direction[sfi, :]),
+                                    uvw=uvw_data,
+                                    refdir=np.rad2deg(zenith_directions[sfi, :].reshape(1, 2)),
+                                    field_ids=map_row_to_zendir,
+                                    sel=sel)
```

Comment:
As above this is done for graph optimization. Each integration on solar system position needs a separate predict because of changing RA DEC